src/pages/api/contact-details/index.test.js

```diff
@@ -0,0 +1,105 @@
+import axios from 'axios'
+import jsonwebtoken from 'jsonwebtoken'
+import { createRequest, createResponse } from 'node-mocks-http'
+import { paramsSerializer } from '@/utils/urls'
+
+import contactDetailsEndpoint from './index.js'
+
+jest.mock('axios')
+
+const {
+  HACKNEY_JWT_SECRET,
+  GSSO_TOKEN_NAME,
+  REPAIRS_SERVICE_API_KEY,
+  AGENTS_GOOGLE_GROUPNAME,
+} = process.env
+
+const CONTACT_DETAILS_API_URL = 'https://test.com/development'
+process.env.CONTACT_DETAILS_API_URL = CONTACT_DETAILS_API_URL
+
+describe('/api/contact-details', () => {
+  beforeEach(() => {
+    axios.create = jest.fn(() => axios)
+
+    axios.mockImplementationOnce(() =>
+      Promise.resolve({
+        status: 200,
+      })
+    )
+  })
+
+  const signedCookie = jsonwebtoken.sign(
+    {
+      name: 'name',
+      email: 'name@example.com',
+      groups: [AGENTS_GOOGLE_GROUPNAME],
+    },
+    HACKNEY_JWT_SECRET
+  )
+
+  const headers = {
+    'Content-Type': 'application/json',
+    'x-api-key': REPAIRS_SERVICE_API_KEY,
+    'x-hackney-user': signedCookie,
+    Authorization: signedCookie,
+  }
+
+  test('DELETE request forwards request to correct endpoint', async () => {
+    const contactId = 'aaa'
+    const personId = 'bbb'
+
+    const req = createRequest({
```

Comment:
I'd change this to createDeleteRequest, otherwise it's ambiguous

Reply:
Not sure how helpful this is here, because in both scenarios, the same endpoint is being called. However, I have extracted this logic into two separate methods: `createDeleteRequest` and `createPatchRequest`. This does make the code more readable, so I will keep it